src/scalar.rs

```diff
@@ -270,12 +270,33 @@ impl Scalar {
     /// require specific bit-patterns when performing scalar
     /// multiplication.
     pub const fn from_bits(bytes: [u8; 32]) -> Scalar {
-        let mut s = Scalar { bytes };
-        // Ensure that s < 2^255 by masking the high bit
-        s.bytes[31] &= 0b0111_1111;
+        let mut bytes:[u8; 32] = bytes;
+        bytes[31] &= 0b0111_1111;
+        Scalar { bytes }
+    }
 
-        s
+    /// Construct a `Scalar` from the low 255 bits of a little-endian 256-bit integer
+    /// `clamping` it's value to be in range
+    ///
+    /// **n ∈ 2^254 + 8\*{0, 1, 2, 3, . . ., 2^251 − 1}**
+    pub const fn from_bits_clamped(bytes: [u8; 32]) -> Scalar {
+        Scalar{ bytes : Scalar::clamp(bytes) }
+    }
```

Comment:
I think there's little value in splitting what you have as `clamp` off into its own function.

```suggestion
    pub const fn from_bits_clamped(mut bytes: [u8; 32]) -> Scalar {
        bytes[0]  &= 0b1111_1000;
        bytes[31] &= 0b0111_1111;
        bytes[31] |= 0b0100_0000;
        Scalar{ bytes }
    }
```

Reply:
two functions merged.